adapters/rest_generic/rest_generic_connect.php

```diff
@@ -95,10 +95,10 @@ public function send($origin, $rest_cmd) {
 		if (count($cmd_list) >1 ) {
 			$rest_path = $cmd_list[1];
 		}
-		
+
 		$headers = "";
 		$auth = "";
-	
+
 		echo("auth_mode= ".$this->auth_mode."\n");
                 echo("auth_header= ".$this->auth_header."\n");
                 echo("key= ".$this->key."\n");
```

Comment:
@n1654 
just got a feedback with the NEC.
they tried to connect to real Northstar and found it needed Basic Auth on top of Bearer.
so we might want to add something like 
`$auth = " -u " . $this->sd_login_entry . ":" . $this->sd_passwd_entry;`
here.
Do you think it works?

Reply:
it already supports basic auth we just need to be able to use both at the same time

Reply:
@Korvuss Is there any way to change auth_mode by MS or by config profiles anyhow?

Reply:
@hiro-ubiqube you can set the Auth_mode in the config variables for the device

Reply:
@Korvuss 
You mean if I set config profiles like below, will MS A use token and MS B&C use Basic auth?
`
config A
 - MS A
 - Template A (AUTH_MODE = token)
config B
 - MS B
 - MS C
 - Template B (AUTH_MODE = Basic)
`